### PR TITLE
Working With APIs lesson: Added link to additional resource section

### PR DESCRIPTION
**Pull request by Rllopez0410**
Updated the additional resource section with its first link to a brief article about the benefits and types of APIs

## Because
The purpose of this pull request is to provide addition knowledge on the benefits and types of APIs that were not discussed in the lesson; such as, release policies, api protocols, and api architectural style.

## This PR
- Included a link in the addition resources section that'll redirect the user to an article with additional information about APIs.

## Issue

Closes #XXXXX

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by cakegod** [comment]
In general, adding new resources should be first discussed before opening a PR.

**Comment by Rllopez0410** [comment]
My apologies. I may have missed that in the contribution docs. Should I cancel the request?

**Comment by cakegod** [comment]
For now, don't cancel it. Wait for our maintainers to further proceed.

**Comment by ManonLef** [comment]
@cakegod [small change to the curriculum are fine to open a PR for](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md#how-to-contribute)

The idea is that we don't want someone to spend a lot of time and resources on big changes only to find out we as maintainers would have rejected the idea in the first place for one reason or another. 

@Rllopez0410 I don't see any harm in adding this but I think there could be more information surrounding the link. For example the points you mentioned this article covers outside what's covered in the lesson.

**Comment by Rllopez0410** [comment]
@ManonLef, that's understandable; thank you.

**Comment by ManonLef** [comment]
Are you willing to make that update @Rllopez0410 ?

**Comment by Rllopez0410** [comment]
@ManonLef yes.

**Comment by Rllopez0410** [comment]
@ManonLef I have updated the information surrounding the link so that readers are better informed about the content within the article.

File: javascript/asynchronous_javascript_and_apis/working_with_apis.md [hunk 1]
```diff
@@ -240,4 +240,4 @@ This section contains questions for you to check your understanding of this less
 
 This section contains helpful links to related content. It isn't required, so consider it supplemental.
 
-- It looks like this lesson doesn't have any additional resources yet. Help us expand this section by contributing to our curriculum.
+- [This article by Ryan Thelin, briefly decribes the external, security, and automated benfits of using an API. It also informs and explains to the reader of the dissimilarity of release policie tiers, in addition to the different API protocols like, Representational State Transfer or also known as (REST) APIs, Remote Procedure Call (RPC), and Simple Object Access Protocol (SOAP). Furthermore, the author of this composition quickly articulates the differences between architectural API styles; for example, service-oriented architecture (SOA) and microservice architecture.](https://www.educative.io/blog/what-is-an-api)
```
Comment (ManonLef) [inline javascript/asynchronous_javascript_and_apis/working_with_apis.md:243]:
```suggestion
- [This article about APIs by Ryan Thelin](https://www.educative.io/blog/what-is-an-api) covers some extra topics like API release policies, API protocols and API architectural styles.
```
Thanks for the update. In my opinion the subscription could be summarised with my above suggestion for example. What do you think? If you agree, feel free to click `commit suggestion` or commit an alternative if you'd like

Comment (ManonLef) [reply javascript/asynchronous_javascript_and_apis/working_with_apis.md:243]: Also, we generally only add a short description about the link between the square brackets so the whole paragraph doesn't turn into a link. I edited that as well :)

**Review by ManonLef — CHANGES_REQUESTED** [review]
I added a comment with a suggested alternative

File: javascript/asynchronous_javascript_and_apis/working_with_apis.md (hunk 1 above)
Comment (Rllopez0410) [reply javascript/asynchronous_javascript_and_apis/working_with_apis.md:243]: This is so much better. Mine was defiantly a bit lengthy. Thank you. 😊

**Review by ManonLef — APPROVED** [review]

**Pull request merged**